apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx

```diff
@@ -19,28 +20,38 @@ const pageComponents = [
 ];
 
 const TeacherApplication = props => {
-  const {accountEmail, userId, schoolId} = props;
+  const {
+    // [MEG] TODO: remove allowPartialSaving prop when experiment is complete (TeacherApps will always have this option)
+    // instead, pass in allowPartialSaving prop to FormController
+    allowPartialSaving,
+    savedFormData,
+    accountEmail,
+    userId,
+    schoolId
+  } = props;
 
   const getInitialData = () => {
+    const dataOnPageLoad =
+      allowPartialSaving && savedFormData && JSON.parse(savedFormData);
+
     // Extract school info saved in sessionStorage, if any
-    let reloadedSchoolId = undefined;
-    if (sessionStorage.getItem(sessionStorageKey)) {
-      const reloadedState = JSON.parse(
-        sessionStorage.getItem(sessionStorageKey)
-      );
-      reloadedSchoolId = reloadedState.data.school;
-    }
+    const reloadedSchoolId = get(
+      JSON.parse(sessionStorage.getItem(sessionStorageKey)),
+      'data.school'
+    );
 
     // Populate additional data from server only if it doesn't override data in sessionStorage
     // (even if value in sessionStorage is null)
     // the FormController will handle loading reloadedSchoolId as an initial value, so return empty otherwise
     if (reloadedSchoolId === undefined && schoolId) {
-      return {school: schoolId};
+      // [MEG] TODO: Check what happens if dataOnPageLoad contains school data that's different from the schoolId
```

Comment:
in this case we would want to override the reloadedSchoolId with the saved school data, because presumably, they got the reloaded data the first time around and put something different in

Reply:
so maybe ignore the reloadedSchoolId when loading from saved data here

Reply:
I think that's what is happening here (not using the `reloadedSchoolId` regardless) ... I'm not sure I understand what you're saying?

Reply:
oh sorry, i meant not to use schoolId
schoolId comes from the server from the user's profile information. and in ```{...dataOnPageLoad, school: schoolId}``` we are overriding the loaded data with that profile data, even though the page load data should be more 'recent'

Reply:
Ah, I see. Then I can just spread `dataOnPageLoad` after the school then ... and it will still overwrite the server data in FormController, which is what we want. Thanks!

Reply:
Actually, this will be more complicated because the keys in `dataOnPageLoad` are different from schoolId ... I'll think about how to best handle this. Thanks, Tim, for the direction!